File: MovieProject/MovieProject.Web/Modules/Northwind/SalesInvoiceItems/SalesInvoiceItemsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace MovieProject.Northwind.Columns;

[ColumnsScript("Northwind.SalesInvoiceItems")]
[BasedOnRow(typeof(SalesInvoiceItemsRow), CheckNames = true)]
public class SalesInvoiceItemsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public DateTime Date { get; set; }
    [EditLink]
    public string Notes { get; set; }
    public string SalesInvoiceNotes { get; set; }
}

File: MovieProject/MovieProject.Web/Modules/ServerTypes/Northwind/SalesInvoiceItemsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SalesInvoiceItemsRow } from "./SalesInvoiceItemsRow";

export interface SalesInvoiceItemsColumns {
    Id: Column<SalesInvoiceItemsRow>;
    Date: Column<SalesInvoiceItemsRow>;
    Notes: Column<SalesInvoiceItemsRow>;
    SalesInvoiceNotes: Column<SalesInvoiceItemsRow>;
}

export class SalesInvoiceItemsColumns extends ColumnsBase<SalesInvoiceItemsRow> {
    static readonly columnsKey = 'Northwind.SalesInvoiceItems';
    static readonly Fields = fieldsProxy<SalesInvoiceItemsColumns>();
}